modules/vault-cluster/main.tf

```diff
@@ -203,6 +203,7 @@ module "security_group_rules" {
 # ---------------------------------------------------------------------------------------------------------------------
 
 resource "aws_iam_instance_profile" "instance_profile" {
+  count       = var.iam_instance_profile_arn == null ? 1 : 0
```

Comment:
Each item that you've added a `count` parameter to is now treated as a list. You need to search for all references to this resource and all the other resources with `count` below and update those references to (a) only reference the resource if it exists and (b) use a list syntax.